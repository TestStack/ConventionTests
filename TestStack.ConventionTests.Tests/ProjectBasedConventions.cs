namespace TestStack.ConventionTests.Tests
{
    using System.Xml.Linq;
    using ApprovalTests;
    using ApprovalTests.Reporters;
    using NSubstitute;
    using NUnit.Framework;
    using TestStack.ConventionTests.Conventions;
    using TestStack.ConventionTests.Helpers;
    using TestStack.ConventionTests.Tests.Properties;

    [TestFixture]
    [UseReporter(typeof(DiffReporter))]
    public class ProjectBasedConventions
    {
        [Test]
        public void ReferencingBinObj()
        {
            // Actual syntax will be (when not testing):
            //
            // Convention.Is<ProjectDoesNotReferenceDllsFromBinOrObjDirectories>(new[] { typeof(ProjectBasedConventions).Assembly });
            //

            var projectProvider = Substitute.For<IProjectProvider>();
            var projectLocator = Substitute.For<IProjectLocator>();
            projectProvider
                .LoadProjectDocument(Arg.Any<string>())
                .Returns(XDocument.Parse(Resources.ProjectFileWithBinReference));

            var convention = new ProjectDoesNotReferenceDllsFromBinOrObjDirectories(projectLocator, projectProvider);

            var exception = Assert.Throws<ConventionFailedException>(() =>
                Convention.Is(convention, new[] { typeof(ProjectBasedConventions).Assembly }));
            Approvals.Verify(exception.Message);
        }

        [Test]
        public void ScriptsNotEmbeddedResources()
        {
            // Actual syntax will be (when not testing):
            //
            // Convention.Is<FilesAreEmbeddedResources>(new[] { typeof(ProjectBasedConventions).Assembly }, i => i.EndsWith(".sql"));
            //

            var projectProvider = Substitute.For<IProjectProvider>();
            var projectLocator = Substitute.For<IProjectLocator>();
            projectProvider
                .LoadProjectDocument(Arg.Any<string>())
                .Returns(XDocument.Parse(Resources.ProjectFileWithInvalidSqlScriptFile));

            var convention = new FilesAreEmbeddedResources(projectLocator, projectProvider);

            var exception = Assert.Throws<ConventionFailedException>(() =>
                Convention.Is(convention, new[] { typeof(ProjectBasedConventions).Assembly }, i => i.EndsWith(".sql")));
            Approvals.Verify(exception.Message);
        }
    }
}