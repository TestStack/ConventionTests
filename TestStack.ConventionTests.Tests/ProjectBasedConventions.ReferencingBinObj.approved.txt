Convention has failing items
	TestStack.ConventionTests.Tests is referencing assemblies in the bin or obj folders:
	bin\Debug\ApprovalTests.dll
